### PR TITLE
Remove the deprecated getters and setters from the generic parameters

### PR DESCRIPTION
**Pull request by jmcarcell**
BEGINRELEASENOTES
- Remove the deprecated getters and setters from the generic parameters

ENDRELEASENOTES

It's the one year birthday this week of the commit that marked them as deprecated so it has been a long time that they have been deprecated and I don't think they are being used much if at all.

### CODE REVIEW

**Review by tmadlener — APPROVED** [review]
I am wondering whether we should also remove the `getXYZMap` functions as they should only be used internally, but we can do that in another PR as well once we have verified we only use them internally.

**Comment by jmcarcell** [comment]
Done, I checked and didn't find any places in the key4hep stack. This makes public the `getMap()` functions

File: include/podio/GenericParameters.h [hunk 1]
```diff
@@ -209,47 +138,6 @@ class GenericParameters {
...
-   * Get the internal string map (necessary for serialization with SIO)
-   */
-  const StringMap& getStringMap() const {
-    return getMap<std::string>();
-  }
-  StringMap& getStringMap() {
-    return getMap<std::string>();
-  }
-
-private:
```
[the remark this answers is not in the document]
Comment (hegner) [reply include/podio/GenericParameters.h:252]: Yap. I would very much prefer that too

[the remark this answers is not in the document]
Comment (jmcarcell) [reply include/podio/GenericParameters.h:252]: Ok now they are friends although I find it not very elegant with the #ifdefs and the extra include, unless there is an easier way of doing it

File: include/podio/GenericParameters.h [hunk 2]
```diff
@@ -12,7 +12,9 @@
 #include <string>
 #include <vector>
 
-#define DEPRECATED_ACCESS [[deprecated("Use templated access functionality")]]
+#if PODIO_ENABLE_SIO
+#include <sio/definitions.h>
+#endif
```
[the remark this answers is not in the document]
Comment (jmcarcell) [reply include/podio/GenericParameters.h]: Now it's done that way

**Review by tmadlener — APPROVED** [review]
Thanks for also addressing the maps. For me this looks good now.

**Comment by jmcarcell** [comment]
This PR makes DD4hep not build for versions before 1.24; that is when the deprecated functions were removed

**Comment by tmadlener** [comment]
Maybe also make a PR to dd4hep itself to increas the min version of podio if it is not already high enough?

**Comment by jmcarcell** [comment]
It's 0.16, but the issue here is with older versions of dd4hep, develop has already removed the deprecated stuff

**Comment by tmadlener** [comment]
It looks like the `getXYZMap` have already escaped into the wild into the wild and are used in DD4hep.

https://github.com/AIDASoft/DD4hep/blob/11d0ab1e17e88ed393b83263db88b2e8f558f9f4/DDDigi/io/DigiIO.cpp#L352-L354

**Pull request merged**